Source/JavaScriptCore/wasm/WasmAirIRGenerator.cpp

```diff
@@ -2220,6 +2191,48 @@ void AirIRGenerator::sanitizeAtomicResult(ExtAtomicOpType op, Type valueType, Tm
     }
 }
 
+void AirIRGenerator::emitWriteBarrier(TypedTmp cell, TypedTmp instanceCell)
+{
+    auto vm = g64();
+    auto cellState = g32();
+    auto threshold = g32();
+
+    BasicBlock* fenceCheckPath = m_code.addBlock();
+    BasicBlock* fencePath = m_code.addBlock();
+    BasicBlock* doSlowPath = m_code.addBlock();
+    BasicBlock* continuation = m_code.addBlock();
+
+    append(Move, Arg::addr(instanceValue(), Instance::offsetOfOwner()), instanceCell);
```

Comment:
Why do we need this? The parameter `instanceCell` should already have instance, right?